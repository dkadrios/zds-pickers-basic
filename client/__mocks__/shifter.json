{
  "accessGranted": true,
  "errorMessage": "",
  "errorVisible": false,
  "exportBuffer": [],
  "exportDialogVisible": false,
  "exportFilename": "zds-shifter-backup.txt",
  "found": true,
  "hardwareTestVisible": true,
  "importDialogVisible": false,
  "importInProcess": false,
  "invalidSettingsFile": null,
  "midiActivityLEDMode": 0,
  "midiInActivity": false,
  "midiOutActivity": false,
  "ready": true,
  "resetInProcess": false,
  "responding": true,
  "searchedForShifter": true,
  "serialMidiOutEnabled": true,
  "showResetDialog": false,
  "testInterfaceFound": false,
  "usbMidiOutEnabled": true
}
