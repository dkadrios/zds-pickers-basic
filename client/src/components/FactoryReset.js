import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogActions from '@material-ui/core/DialogActions'
import LinearProgress from '@material-ui/core/LinearProgress'
import { shifterShape } from '../core/shapes'
import Dialog from './Dialog'

const FactoryReset = (props) => {
  const { confirmFactoryReset, performFactoryReset, shifter } = props
  const { resetInProcess, showResetDialog } = shifter

  return (
    <>
      <Dialog
        onClose={/* istanbul ignore next */ () => confirmFactoryReset(false)}
        open={showResetDialog}
      >
        <DialogTitle>Confirm Factory Reset</DialogTitle>
        <DialogContent>
          <DialogContentText>
            <b>Sure to perform reset?</b>
            <br />
            <br />
            This will remove all shift groups and return the foot switches to their factory defaults.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button
            onClick={() => confirmFactoryReset(false)}
            tag="btnCancel"
          >
            Cancel
          </Button>
          <Button
            autoFocus
            color="primary"
            onClick={() => performFactoryReset(true)}
            tag="btnOk"
            variant="contained"
          >
            Perform Reset
          </Button>
        </DialogActions>
      </Dialog>

      <Dialog open={resetInProcess}>
        <DialogTitle>Factory Reset In Progress</DialogTitle>
        <DialogContent>
          <DialogContentText>Please wait...</DialogContentText>
          <LinearProgress />
        </DialogContent>
      </Dialog>
    </>
  )
}

FactoryReset.propTypes = {
  shifter: shifterShape.isRequired,
  confirmFactoryReset: PropTypes.func.isRequired,
  performFactoryReset: PropTypes.func.isRequired,
}

export default FactoryReset
