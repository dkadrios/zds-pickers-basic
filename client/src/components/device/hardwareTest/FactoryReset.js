/* eslint-disable react/no-unescaped-entities */
import { hardwareTestShape } from '../../../core/shapes'

const FactoryReset = ({ hardwareTest: { performingReset } }) => performingReset ? ( //
  <h3>Performing factory reset...</h3>
) : (
  <>
    <h3>Reset Complete.</h3>
    <h4>Click 'Next' to create test shift rules...</h4>
    <h4>Both foot switches will also be enabled and the LEDs should illuminate.</h4>
  </>
)

FactoryReset.propTypes = {
  hardwareTest: hardwareTestShape.isRequired,
}

export default FactoryReset
