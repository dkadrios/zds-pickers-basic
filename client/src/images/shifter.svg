<svg x="0" y="0" width="240" height="57" viewBox="0, 0, 480, 114">
  <g id="zds_text">
    <g>
      <g>
        <path
          d="M8.442,37.216 L75.696,37.216 C80.083,37.216 83.639,40.265 83.639,44.027 L83.639,66.448 C83.639,70.209 80.083,73.259 75.696,73.259 L8.442,73.259 C4.056,73.259 0.5,70.209 0.5,66.448 L0.5,44.027 C0.5,40.265 4.056,37.216 8.442,37.216 z"
          fill="#7e0018"
        />
        <path
          d="M8.442,37.216 L75.696,37.216 C80.083,37.216 83.639,40.265 83.639,44.027 L83.639,66.448 C83.639,70.209 80.083,73.259 75.696,73.259 L8.442,73.259 C4.056,73.259 0.5,70.209 0.5,66.448 L0.5,44.027 C0.5,40.265 4.056,37.216 8.442,37.216 z"
          fillOpacity="0"
          stroke="#7e0018"
          strokeWidth="1"
        />
      </g>
      <path
        d="M31.017,67.192 L30.513,67.012 L30.513,66.76 Q30.441,65.968 30.099,65.68 Q29.757,65.392 28.749,65.392 L14.925,65.392 L26.697,46.6 L17.049,46.6 Q16.041,46.6 15.717,46.888 Q15.393,47.176 15.321,47.968 L15.321,48.22 L14.817,48.436 L14.817,48.112 Q14.817,47.788 14.853,47.176 Q14.889,46.564 14.889,46.384 L14.889,44.584 Q14.889,44.368 14.853,43.756 Q14.817,43.144 14.817,42.856 L14.817,42.568 L15.321,42.748 L15.321,43 Q15.393,43.792 15.717,44.08 Q16.041,44.368 17.049,44.368 L30.801,44.368 L19.065,63.16 L28.749,63.16 Q29.757,63.16 30.099,62.872 Q30.441,62.584 30.513,61.792 L30.513,61.54 L31.017,61.324 L31.017,61.648 Q31.017,61.936 30.963,62.548 Q30.909,63.16 30.909,63.376 L30.909,65.176 Q30.909,65.356 30.963,65.986 Q31.017,66.616 31.017,66.904 z"
        fill="#FFFFFF"
      />
      <path
        d="M53.585,54.808 Q53.585,59.308 50.453,62.368 Q47.321,65.428 42.785,65.428 Q41.993,65.428 40.697,65.428 Q39.401,65.392 39.329,65.392 L36.737,65.392 Q36.557,65.392 35.945,65.428 Q35.333,65.464 35.009,65.464 L34.721,65.464 L34.901,64.996 L35.153,64.996 Q35.945,64.924 36.233,64.582 Q36.521,64.24 36.521,63.232 L36.521,46.528 Q36.521,45.52 36.233,45.178 Q35.945,44.836 35.153,44.764 L34.901,44.764 L34.721,44.296 L35.009,44.296 Q35.333,44.296 35.945,44.332 Q36.557,44.368 36.737,44.368 L38.897,44.368 Q39.329,44.368 40.589,44.332 Q41.849,44.296 42.785,44.296 Q47.249,44.296 50.417,47.338 Q53.585,50.38 53.585,54.808 z M51.317,54.916 Q51.317,51.388 48.851,48.886 Q46.385,46.384 42.893,46.384 L38.825,46.384 L38.825,63.376 L42.893,63.376 Q46.385,63.376 48.851,60.892 Q51.317,58.408 51.317,54.916 z"
        fill="#FFFFFF"
      />
      <path
        d="M70.537,59.884 Q70.537,62.512 68.521,64.186 Q66.505,65.86 63.697,65.86 Q60.781,65.86 58.945,63.88 Q57.721,62.548 57.433,61.108 L60.385,59.812 L60.601,60.172 Q59.845,60.64 59.845,61.144 Q59.845,61.864 60.853,62.728 Q62.077,63.808 63.805,63.808 Q65.821,63.808 67.063,62.656 Q68.305,61.504 68.305,59.848 Q68.305,57.904 65.209,56.212 L61.897,54.412 Q59.809,53.26 58.981,52 Q58.297,50.92 58.297,49.336 Q58.297,47.068 59.989,45.592 Q61.681,44.116 63.985,44.116 Q68.017,44.116 69.817,47.608 L67.369,49.804 L67.081,49.336 Q67.657,48.796 67.657,48.328 Q67.657,47.536 66.595,46.834 Q65.533,46.132 64.201,46.132 Q62.365,46.132 61.393,47.14 Q60.493,48.004 60.493,49.336 Q60.493,51.064 63.193,52.504 L67.081,54.592 Q70.537,56.464 70.537,59.884 z"
        fill="#FFFFFF"
      />
    </g>
  </g>
  <g id="shifter_txt">
    <g>
      <path
        d="M119.324,101.088 Q117.884,105.552 112.484,108.936 Q107.084,112.32 102.188,112.32 L21.26,112.32 L32.204,78.624 L86.204,78.624 Q99.596,78.624 103.196,67.392 L110.54,44.928 Q111.26,42.768 111.26,40.896 Q111.26,33.696 100.748,33.696 L73.82,33.696 Q63.452,33.696 63.452,25.632 Q63.452,24.048 63.884,22.464 L67.628,11.232 Q68.924,7.2 73.964,3.6 Q79.004,0 84.764,0 L125.228,0 L121.484,11.232 L94.556,11.232 L90.956,22.464 L131.276,22.464 Q141.788,22.464 141.788,29.952 Q141.788,31.68 141.212,33.696 z"
        fill="#000000"
      />
      <path
        d="M195.212,67.392 L181.82,67.392 L192.764,33.696 L179.228,33.696 L168.284,67.392 L141.356,67.392 L163.244,0 L190.172,0 L182.828,22.464 L196.364,22.464 Q206.732,22.464 206.732,30.096 Q206.732,31.824 206.156,33.696 z"
        fill="#000000"
      />
      <path
        d="M254.684,11.232 L227.756,11.232 L231.356,0 L258.428,0 z M236.54,67.392 L209.468,67.392 L224.156,22.464 L251.084,22.464 z"
        fill="#000000"
      />
      <path
        d="M298.172,44.928 L284.78,44.928 L277.436,67.392 L250.508,67.392 L268.652,11.232 Q270.38,6.048 275.708,2.88 Q280.46,0 285.788,0 L312.86,0 L309.116,11.232 L295.724,11.232 L288.38,33.696 L301.916,33.696 z"
        fill="#000000"
      />
      <path
        d="M343.388,67.392 L316.46,67.392 Q311.996,67.392 309.044,65.232 Q306.092,63.072 306.092,59.328 Q306.092,57.744 306.524,56.16 L324.812,0 L351.74,0 L344.54,22.464 L357.932,22.464 L354.332,33.696 L340.796,33.696 L337.196,44.928 L350.588,44.928 z"
        fill="#000000"
      />
      <path
        d="M410.636,67.392 L370.316,67.392 Q366.14,67.392 363.26,65.232 Q359.948,62.784 359.948,59.04 Q359.948,57.6 360.38,56.16 L367.724,33.696 Q369.164,29.376 374.42,25.92 Q379.676,22.464 384.86,22.464 L416.972,22.464 Q425.324,22.464 425.324,27.648 Q425.324,31.104 421.436,33.696 L404.444,44.928 L417.98,44.928 z M408.188,33.696 L394.652,33.696 L391.052,44.928 z"
        fill="#000000"
      />
      <path
        d="M476.588,33.696 L463.196,33.696 L452.252,67.392 L425.324,67.392 L436.268,33.696 Q437.708,28.944 443.324,25.488 Q448.508,22.464 453.404,22.464 L480.332,22.464 z"
        fill="#000000"
      />
      <g>
        <path
          d="M138.673,79.309 L460.926,79.309 L450.608,113.337 L128.355,113.337 L138.673,79.309 z"
          fill="#000000"
        />
        <path
          d="M138.673,79.309 L460.926,79.309 L450.608,113.337 L128.355,113.337 L138.673,79.309 z"
          fillOpacity="0"
          stroke="#000000"
          strokeWidth="1"
        />
      </g>
    </g>
  </g>
</svg>
