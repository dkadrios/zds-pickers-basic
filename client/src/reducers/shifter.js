/* eslint-disable no-bitwise */
import { submit } from 'redux-form'
import { createReducer } from '../utils'
import { ACTIVITY_LED_MODE_ALWAYS_OFF } from '../midi/sysex'
import actionTypes from './actionTypes'

export const actions = {
  searchedForShifter: () => ({
    type: actionTypes.SEARCHED_FOR_SHIFTER,
  }),

  shifterFound: () => ({
    type: actionTypes.SHIFTER_FOUND,
  }),

  shifterMissing: () => ({
    type: actionTypes.SHIFTER_MISSING,
  }),

  testInterfaceFound: () => ({
    type: actionTypes.TEST_INTERFACE_FOUND,
  }),

  testInterfaceMissing: () => ({
    type: actionTypes.TEST_INTERFACE_MISSING,
  }),

  notResponding: () => ({
    type: actionTypes.NOT_RESPONDING,
  }),

  midiInActivityChanged: midiInActivity => ({
    type: actionTypes.MIDI_IN_ACTIVITY,
    midiInActivity,
  }),

  midiOutActivityChanged: midiOutActivity => ({
    type: actionTypes.MIDI_OUT_ACTIVITY,
    midiOutActivity,
  }),

  confirmFactoryReset: showResetDialog => ({
    type: actionTypes.CONFIRM_FACTORY_RESET,
    showResetDialog,
  }),

  showExportDialog: exportDialogVisible => ({
    type: actionTypes.SHOW_EXPORT_DIALOG,
    exportDialogVisible,
  }),

  submitExportForm: () => (dispatch) => {
    dispatch(submit('exportSettingsForm'))
  },

  exportSettings: ({ exportFilename }) => ({
    type: actionTypes.EXPORT_SETTINGS,
    exportDialogVisible: false,
    exportFilename,
  }),

  receivedExportPacket: packet => ({
    type: actionTypes.EXPORT_SETTINGS_PACKET,
    packet,
  }),

  showImportDialog: importDialogVisible => ({
    type: actionTypes.SHOW_IMPORT_DIALOG,
    importDialogVisible,
  }),

  submitImportForm: () => (dispatch) => {
    dispatch(submit('importSettingsForm'))
  },

  importSettings: File => ({
    type: actionTypes.IMPORT_SETTINGS,
    importDialogVisible: false,
    File,
  }),

  settingsFileInvalid: reason => ({
    type: actionTypes.INVALID_SETTINGS_FILE,
    invalidSettingsFile: reason,
  }),

  acknowledgeInvalidFile: () => ({
    type: actionTypes.INVALID_SETTINGS_FILE_ACK,
  }),

  restart: () => ({
    type: actionTypes.RESTART,
  }),

  performFactoryReset: restartToo => ({
    type: actionTypes.FACTORY_RESET,
    restartToo,
  }),

  setFlags: (midiActivityLEDMode, serialMidiOutEnabled, usbMidiOutEnabled) => ({
    type: actionTypes.SET_FLAGS,
    midiActivityLEDMode,
    serialMidiOutEnabled,
    usbMidiOutEnabled,
  }),

  receivedAvailability: ready => ({
    type: actionTypes.RECEIVED_AVAILABILITY,
    ready,
  }),

  reportError: errorMessage => ({
    type: actionTypes.REPORT_ERROR,
    errorMessage,
  }),

  dismissError: () => ({
    type: actionTypes.DISMISS_ERROR,
  }),

  showHardwareTestDialog: () => ({
    type: actionTypes.SHOW_HARDWARE_TEST_DIALOG,
  }),

  hideHardwareTestDialog: () => ({
    type: actionTypes.HIDE_HARDWARE_TEST_DIALOG,
  }),
}

export const defaultState = {
  accessGranted: false,
  errorMessage: '',
  errorVisible: false,
  exportBuffer: [],
  exportDialogVisible: false,
  exportFilename: localStorage.getItem('exportFilename') || 'zds-shifter-backup.txt',
  found: false,
  hardwareTestVisible: false,
  importDialogVisible: false,
  importInProcess: false,
  invalidSettingsFile: null,
  midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_OFF,
  midiInActivity: false,
  midiOutActivity: false,
  ready: false,
  resetInProcess: false,
  responding: true, // Assume responding until proven otherwise
  searchedForShifter: false,
  serialMidiOutEnabled: true,
  showResetDialog: false,
  testInterfaceFound: false,
  usbMidiOutEnabled: true,
}

const errorReported = (state, { errorMessage }) => ({
  ...state,
  errorMessage,
  errorVisible: true,
})

const dismissError = state => ({
  ...state,
  errorVisible: false,
})

const searchedForShifter = state => ({
  ...state,
  searchedForShifter: true,
})

const shifterFound = state => ({
  ...state,
  found: true,
  showResetDialog: false,
  resetInProcess: false,
})

const shifterMissing = state => ({
  ...state,
  found: false,
  responding: true, // Assume responding until proven otherwise
  showResetDialog: false,
  resetInProcess: false,
})

const testInterfaceFound = state => ({
  ...state,
  testInterfaceFound: true,
})

const testInterfaceMissing = state => ({
  ...state,
  testInterfaceFound: false,
})

const toggleHardwareTestDialog = hardwareTestVisible => state => ({
  ...state,
  hardwareTestVisible,
})

const notResponding = state => ({
  ...state,
  responding: false,
})

const responding = state => ({
  ...state,
  responding: true,
})

const receivedDeviceList = state => ({
  ...state,
  accessGranted: true, // assume we have access if device list was received
})

const midiInActivityChanged = (state, { midiInActivity }) => ({
  ...state,
  midiInActivity,
})

const midiOutActivityChanged = (state, { midiOutActivity }) => ({
  ...state,
  midiOutActivity,
})

const confirmFactoryReset = (state, { showResetDialog }) => ({
  ...state,
  showResetDialog,
})

const resetBeingPerformed = (state, { restartToo }) => ({
  ...state,
  showResetDialog: false,
  resetInProcess: restartToo,
})

const factoryResetPerformed = (state, { packet }) => {
  let { midiActivityLEDMode, serialMidiOutEnabled, usbMidiOutEnabled } = state

  if (packet) {
    const [numControls, ...rest] = packet

    // `rest` has "active" and "lit states" for all `numControls` (6) inputs, followed
    // by the "active" state for the four groups, then finally the LED mode flag
    midiActivityLEDMode = rest[numControls + numControls + 4] & 3
    serialMidiOutEnabled = Boolean(rest[numControls + numControls + 4] & 4)
    usbMidiOutEnabled = Boolean(rest[numControls + numControls + 4] & 8)
  }

  return {
    ...state,
    showResetDialog: false,
    resetInProcess: false,
    midiActivityLEDMode,
    serialMidiOutEnabled,
    usbMidiOutEnabled,
  }
}

const showExportDialog = (state, { exportDialogVisible = true }) => ({
  ...state,
  exportDialogVisible,
  exportBuffer: [],
})

const exportSettings = (state, { exportDialogVisible, exportFilename }) => {
  localStorage.setItem('exportFilename', exportFilename)
  return {
    ...state,
    exportDialogVisible,
    exportFilename,
  }
}

const receivedExportPacket = (state, { packet }) => ({
  ...state,
  exportBuffer: [...state.exportBuffer, ...packet],
})

const showImportDialog = (state, { importDialogVisible = true }) => ({
  ...state,
  importDialogVisible,
})

const importSettings = state => ({
  ...state,
  importDialogVisible: false,
  importInProcess: true,
})

const importComplete = state => ({
  ...state,
  importInProcess: false,
})

const settingsFileInvalid = (state, { invalidSettingsFile }) => ({
  ...state,
  invalidSettingsFile,
})

const acknowledgedInvalidFile = state => ({
  ...state,
  invalidSettingsFile: null,
})

const receivedAvailability = (state, { ready }) => ({
  ...state,
  ready,
})

export default createReducer(defaultState, {
  [actionTypes.REPORT_ERROR]: errorReported,
  [actionTypes.DISMISS_ERROR]: dismissError,
  [actionTypes.SEARCHED_FOR_SHIFTER]: searchedForShifter,
  [actionTypes.SHIFTER_FOUND]: shifterFound,
  [actionTypes.SHIFTER_MISSING]: shifterMissing,
  [actionTypes.TEST_INTERFACE_FOUND]: testInterfaceFound,
  [actionTypes.TEST_INTERFACE_MISSING]: testInterfaceMissing,
  [actionTypes.SHOW_HARDWARE_TEST_DIALOG]: toggleHardwareTestDialog(true),
  [actionTypes.HIDE_HARDWARE_TEST_DIALOG]: toggleHardwareTestDialog(false),
  'redux-midi/midi/RECEIVE_DEVICE_LIST': receivedDeviceList,
  [actionTypes.MIDI_IN_ACTIVITY]: midiInActivityChanged,
  [actionTypes.MIDI_OUT_ACTIVITY]: midiOutActivityChanged,
  [actionTypes.NOT_RESPONDING]: notResponding,
  [actionTypes.RECEIVED_VERSION]: responding,

  [actionTypes.CONFIRM_FACTORY_RESET]: confirmFactoryReset,
  [actionTypes.FACTORY_RESET]: resetBeingPerformed,
  /* Instead of catching 'FACTORY_RESET', watch for the actual result
   * of the reset, which is when it exports internal state.
   */
  [actionTypes.RECEIVED_STATE]: factoryResetPerformed,
  [actionTypes.SHOW_EXPORT_DIALOG]: showExportDialog,
  [actionTypes.EXPORT_SETTINGS]: exportSettings,
  [actionTypes.SHOW_IMPORT_DIALOG]: showImportDialog,
  [actionTypes.IMPORT_SETTINGS]: importSettings,
  /* When importing, we know it's done for sure when we receive the updated groups.
   */
  [actionTypes.RECEIVED_GROUPS]: importComplete,
  [actionTypes.INVALID_SETTINGS_FILE]: settingsFileInvalid,
  [actionTypes.INVALID_SETTINGS_FILE_ACK]: acknowledgedInvalidFile,
  [actionTypes.RECEIVED_AVAILABILITY]: receivedAvailability,
  [actionTypes.EXPORT_SETTINGS_PACKET]: receivedExportPacket,
})
