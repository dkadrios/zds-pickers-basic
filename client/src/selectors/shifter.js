import { createSelector } from 'reselect'
import { compose } from 'redux'
import { get } from '../core/fp/objects'
import { not } from '../core/fp/utils'
import { stateShifter } from '.'

export const doesShifterExist = createSelector(
  stateShifter,
  get('found'),
)

export const isDisabled = createSelector(
  stateShifter,
  compose(
    not,
    get('ready'),
  ),
)

export default undefined
