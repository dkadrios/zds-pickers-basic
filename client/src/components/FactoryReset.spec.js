import shifter from '../../__mocks__/shifter'
import FactoryReset from './FactoryReset'

const confirmFactoryReset = jest.fn()
const performFactoryReset = jest.fn()

const props = {
  shifter,
  confirmFactoryReset,
  performFactoryReset,
}

describe('FactoryReset tests', () => {
  beforeEach(jest.resetAllMocks)

  it('renders correctly', () => {
    shallowExpect(FactoryReset)(props).toMatchSnapshot()
  })

  it('cancels when clicked', () => {
    const wrapper = shallow(<FactoryReset {...props} />)
    wrapper
      .find('WithStyles(Button)')
      .filter('[tag="btnCancel"]')
      .simulate('click')
    expect(confirmFactoryReset).toHaveBeenCalledWith(false)
  })

  it('proceeds when clicked', () => {
    const wrapper = shallow(<FactoryReset {...props} />)
    wrapper
      .find('WithStyles(Button)')
      .filter('[tag="btnOk"]')
      .simulate('click')
    expect(performFactoryReset).toHaveBeenCalledWith(true)
  })
})
