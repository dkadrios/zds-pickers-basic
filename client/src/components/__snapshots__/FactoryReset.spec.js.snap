// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactoryReset tests renders correctly 1`] = `
<Fragment>
  <Dialog
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogTitle)>
      Confirm Factory Reset
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)>
        <b>
          Sure to perform reset?
        </b>
        <br />
        <br />
        This will remove all shift groups and return the foot switches to their factory defaults.
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        onClick={[Function]}
        tag="btnCancel"
      >
        Cancel
      </WithStyles(Button)>
      <WithStyles(Button)
        autoFocus={true}
        color="primary"
        onClick={[Function]}
        tag="btnOk"
        variant="contained"
      >
        Perform Reset
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </Dialog>
  <Dialog
    open={false}
  >
    <WithStyles(DialogTitle)>
      Factory Reset In Progress
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)>
        Please wait...
      </WithStyles(DialogContentText)>
      <WithStyles(LinearProgress) />
    </WithStyles(DialogContent)>
  </Dialog>
</Fragment>
`;
